### PR TITLE
Fix mac Builds

### PR DESCRIPTION
**Pull request by casperlamboo**
Since the pyqt 6.6 upgrade the mac builds were failing. Main fix was to upgrade pyinstaller (which allowed us to remove a whole lot of ugly ductyping). 

In the new packaged app the file structure is a bit different for mac, so had to change some paths.

Also see Uranum PR: https://github.com/Ultimaker/Uranium/pull/936/files

CURA-11483

### CODE REVIEW

File: UltiMaker-Cura.spec.jinja [hunk 1]
```diff
@@ -70,188 +70,7 @@ coll = COLLECT(
...
-        try:
-            osxutils.sign_binary(self.name, self.codesign_identity, self.entitlements_file, deep = True)
-        except Exception as e:
-            logger.warning(f"Error while signing the bundle: {e}")
-            logger.warning("You will need to sign the bundle manually!")
-
-        logger.info(f"Building BUNDLE {self.tocbasename} completed successfully.")
-
-app = UMBUNDLE(
+app = BUNDLE(
```
Comment (jellespijker) [inline UltiMaker-Cura.spec.jinja:73]: If we don't need the duck typing anymore, then we probably can skip this whole conditional jinja block and go for the out-of-the-box solution

File: conanfile.py [hunk 2]
```diff
@@ -163,8 +163,10 @@ def _conan_installs(self):
     def _python_installs(self):
         python_installs = {}
 
+        python_executable = "python3" if self.settings.os == "Macos" else "python"
+
         # list of python installs
-        python_ins_cmd = f"python -c \"import pkg_resources; print(';'.join([(s.key+','+ s.version) for s in pkg_resources.working_set]))\""
+        python_ins_cmd = f"{python_executable} -c \"import pkg_resources; print(';'.join([(s.key+','+ s.version) for s in pkg_resources.working_set]))\""
```
Comment (jellespijker) [inline conanfile.py:169]: This shouldn't be needed anymore with my recent changes. If it is still needed then apply the environment context to this, see:

- https://github.com/Ultimaker/conan-config/pull/10
- https://github.com/Ultimaker/conan-config/pull/9
- https://github.com/Ultimaker/Cura/pull/18138

File: cura/CuraApplication.py [hunk 3]
```diff
@@ -389,6 +391,7 @@ def __addExpectedResourceDirsAndSearchPaths(self):
 
             # venv site-packages
             Resources.addSearchPath(os.path.join(app_root, "..", "share", "cura", "resources"))
+            Resources.addSearchPath(os.path.join(app_root, "..", "Resources", "share", "cura", "resources"))
```
Comment (jellespijker) [inline cura/CuraApplication.py:394]: Yeah sure, just add all those extra paths ;-)

File: requirements-dev.txt [hunk 4]
```diff
@@ -1,5 +1,5 @@
 pytest
-pyinstaller==5.8.0
+pyinstaller==6.3.0
```
Comment (jellespijker) [inline requirements-dev.txt:2]: If we're using the latest version then I would suggest to unpin the version

**Review by jellespijker — COMMENTED** [review]
Just some adhoc remarks, I know it is not ready yet

File: UltiMaker-Cura.spec.jinja (hunk 1 above)
Comment (casperlamboo) [reply UltiMaker-Cura.spec.jinja:73]: 🦆

File: cura/CuraApplication.py [hunk 5]
```diff
@@ -373,9 +373,17 @@ def __addExpectedResourceDirsAndSearchPaths(self):
...
-        Resources.addSecureSearchPath(os.path.join(self._app_install_dir, "share", "cura", "resources"))
+        if platform.system() == "Darwin":
+            Resources.addSecureSearchPath(os.path.join(app_root, "Resources", "share", "cura", "resources"))
+            Resources.addSecureSearchPath(os.path.join(app_root, "..", "Resources", "share", "cura", "resources"))
+            Resources.addSecureSearchPath(
+                os.path.join(self._app_install_dir, "..", "Resources", "share", "cura", "resources"))
+        else:
+            Resources.addSecureSearchPath(os.path.join(app_root, "share", "cura", "resources"))
+            Resources.addSecureSearchPath(os.path.join(self._app_install_dir, "share", "cura", "resources"))
+            Resources.addSecureSearchPath(os.path.join(app_root, "Resources", "share", "cura", "resources"))
```
Comment (wawanbreton) [inline cura/CuraApplication.py:385]: This one is duplicated in the `if` and in the `else`, so it could be outside (or removed somewhere)

**Review by wawanbreton — COMMENTED** [review]
Looks good, less code = less bugs !

File: cura/CuraApplication.py (hunk 5 above)
Comment (casperlamboo) [reply cura/CuraApplication.py:385]: oh yea, you're right. Let me change this

Comment (casperlamboo) [reply cura/CuraApplication.py]: https://github.com/Ultimaker/Cura/pull/18188/commits/d8cc99bf0deb951d32059e47b7690a0c76babfcb

**Pull request merged**